test/test_helper.rb

```diff
@@ -4,6 +4,14 @@
 # require "pp"
 require "declarative/testing"
 
+# Helper method to normalize hash inspect format for cross-Ruby version compatibility
+# Ruby 3.4 changed from {:key => value} to {key: value} in inspect output
+def normalize_inspect(inspect_string)
+  inspect_string
+    .gsub(/" => /, '"=>')                      # Remove spaces around => for string keys: "key" => value → "key"=>value
```

Comment:
Cool, we have this https://github.com/trailblazer/trailblazer-core-utils/blob/master/lib/trailblazer/core/utils/inspect.rb#L14 which we could use here, but I will test your version! :green_heart: 

Would you mind making the tests `assert_equal asserted, expected`?

Reply:
@apotonick Done! 😃